src/GameStateMenu.cpp

```diff
@@ -10,143 +10,509 @@
+                        textInput[i].pop_back();
+                    }
+
+                    //Handle copy
+                    else if (keyCode == SDLK_c && SDL_GetModState() & KMOD_CTRL) {
+                        SDL_SetClipboardText( textInput[i].c_str() );
+                    }
+
+                    //Handle paste
+                    else if (keyCode == SDLK_v && SDL_GetModState() & KMOD_CTRL) {
+                        textInput[i] = SDL_GetClipboardText();
+                    }
+                }
+            }
+        	break;
+        case SDL_TEXTINPUT:
+            for (size_t i = 0; i < NUM_TEXT_FIELDS; i++) {
+                if (activeTextbox[i]) {
+                    //Ensures that a copy or paste is not occurring
+                    if (!((event.text.text[ 0 ] == 'c' || event.text.text[ 0 ] == 'C') &&
+                          (event.text.text[ 0 ] == 'v' || event.text.text[ 0 ] == 'V') &&
+                           SDL_GetModState() & KMOD_CTRL)) {
+                        //Update the string int displayed int he textbox
+                        if (textInput[i].length() < maxLength) {
+                            textInput[i] += event.text.text;
+                        }
+                    }
+                }
             }
             break;
-        case SDL_QUIT:
-            play = false;
-            break;
-          default:
-            break;
-        }
-    }
+      	case SDL_KEYUP: //Do nothing on key release
+       		switch ( event.key.keysym.sym ) {
+				default:
+	               	break;
+			}
+        	break;
+        case SDL_MOUSEMOTION:
+        	x = event.motion.x;
+        	y = event.motion.y;
+        	for (size_t i = 0; i < NUM_MENU_ITEMS; i++) {
+                //Check if mouse is above one of the menu uptions
+        		if (x >= menuItemPos[i].x && x <= menuItemPos[i].x + menuTextTextures[i].getWidth() &&
+        			y >= menuItemPos[i].y && y <= menuItemPos[i].y + menuTextTextures[i].getHeight()) {
+                    selected[i] = true;  //Activate the button
+        		} else {
+                    selected[i] = false; //deactivate the button
+        		}
+        	}
+        	break;
+        case SDL_MOUSEBUTTONDOWN:
+        	x = event.button.x;
+        	y = event.button.y;
+        	for (size_t i = 0; i < NUM_MENU_ITEMS; i++) {
+                if (selected[i]) {
+                    update(i);
+        			play = false;
+                }
+        	}
+        	for (size_t i = 0; i < NUM_TEXT_FIELDS; i++) {
+                //Check if the mouse was clicked inside the textbox area
+        		if (x >= textboxPos[i].x && x <= textboxPos[i].x + textboxPos[i].w &&
+        			y >= textboxPos[i].y && y <= textboxPos[i].y + textboxPos[i].h) {
+                    activeTextbox[i] = true; //Activate the textbox to enable typing
+                    //If default string is detected, delete it
+                    if (!strcmp(textInput[i].c_str(), defaultText[i].c_str())) {
```

Comment:
converting to c style strings then using strcmp works but doesnt utilize what was already made for strings see [string compare](http://www.cplusplus.com/reference/string/string/compare/)